assembly/src/parsers/mod.rs

```diff
@@ -152,3 +152,32 @@ fn parse_element_param(op: &Token, param_idx: usize) -> Result<BaseElement, Asse
 
     Ok(BaseElement::new(result))
 }
+
+/// This is a helper function that validates the length of the assembly io instruction and returns
+/// an error if params are missing or there are extras.
+///
+/// `instr_parts` expects the number of non-parameter parts in the instruction, e.g. 2 for the
+/// "mem.pop" instruction. `min_params` and `max_params` expect the minimum and maximum number of
+/// parameters accepted by the operation respectively.
+///
+/// # Errors
+///
+/// This function will return an AssemblyError if the instruction part of the operation is
+/// too short or if too many or too few parameters are provided.
+fn validate_op_len(
+    op: &Token,
+    instr_parts: usize,
+    min_params: usize,
+    max_params: usize,
+) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        missing_instr if missing_instr < instr_parts => Err(AssemblyError::invalid_op(op)),
+        missing_param if missing_param < instr_parts + min_params => {
+            Err(AssemblyError::missing_param(op))
+        }
+        extra_param if extra_param > instr_parts + max_params => {
+            Err(AssemblyError::extra_param(op))
+        }
+        _ => Ok(()),
+    }
```

Comment:
I wonder if using simple `if` statements might be a more clear way to express this. `match` is usually more suited when matching against a single value.